ui/src/pages/overview/branch-listing.tsx

```diff
@@ -0,0 +1,85 @@
+import { useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useForm } from '@principlestudios/react-jotai-forms';
+import { useSuspenseQuery } from '@tanstack/react-query';
+import { useAtomValue } from 'jotai';
+import { z } from 'zod';
+import { BulletList, Link, Section } from '../../components/common';
+import { TextField } from '../../components/form/text-field';
+import { queries } from '../../utils/api/queries';
+import type { StandardField } from '../../components/form/FieldProps';
+
+const branchListingSearchSchema = z.object({
+	branchName: z.string(),
+});
+
+export function useBranchListing() {
+	const { t } = useTranslation('branch-listing', { keyPrefix: 'form' });
+	const form = useForm({
+		defaultValue: { branchName: '' },
+		schema: branchListingSearchSchema,
+		translation: t,
+		fields: {
+			branchName: ['branchName'],
+		},
+		preSubmit: 'all',
+	});
+	const response = useSuspenseQuery(queries.getUpstreamData).data;
+
+	const BranchListing = useCallback(
+		function BranchListing() {
+			const branchName = useAtomValue(form.fields.branchName.value);
+			if (branchName.length === 0) return null;
+			const branches = response
+				.filter((r) => r.name.includes(branchName))
+				.map((r) => r.name)
+				.filter((r, i) => i < 100);
```

Comment:
Nit: it's not obvious (to me) why the value `100` is chosen here.  Consider naming this magic number to make the code self-documenting.

Reply:
It was literally a magic number chosen at random to prevent too many items. Should be replaced in a future branch as the UI is improved. (Accepting contributions!)

Reply:
okay cool, that makes sense. how about this then?

`.filter((r, i) => i < ARBITRARY_MAX_BRANCHES);`